tutorials/vbpr_text.ipynb

```diff
@@ -0,0 +1,214 @@
+    "ml_100k = movielens.load_100k(reader=Reader(item_set=movie_ids)) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Cross modality\n",
+    "\n",
+    "To get vector representations from text data, we build a `TextModality` using our corpus and corresponding ids. We also need to supply a `Tokenizer` for text splitting, in this case tokens are seperated by `\\tab` character. We limit the maximum size of vocabulary to 5000, which also means the dimension of our vector space cannot go higher."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "item_text_modality = TextModality(corpus=plots, ids=movie_ids, \n",
+    "                                  tokenizer=BaseTokenizer(sep='\\t', stop_words='english'),\n",
+    "                                  max_vocab=5000, max_doc_freq=0.5).build()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next step is to create an `ImageModality`, which is use by VBPR, using our text representations. In this case, we take the word-count matrix to substitute for visual features."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "features = item_text_modality.count_matrix.A\n",
+    "item_image_modality = ImageModality(features=features, ids=movie_ids)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In Cornac, every model relies on the modality for which it was designed for (i.e., visual recommendation algorithms always work with `ImageModality`). This ensures consistency with models' original assumptions, and helps us avoid confusions regarding which modality to use when integrating a new recommender model.\n",
+    "\n",
+    "## Experiment\n",
+    "\n",
+    "We employ the `RatioSplit` evaluation method to split the rating data. The `item_image_modality` is also supplied here for later usage by the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ratio_split = RatioSplit(data=ml_100k, test_size=0.9,\n",
+    "                         item_image=item_image_modality,\n",
+    "                         exclude_unknowns=True, \n",
+    "                         verbose=True, seed=123)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We are now ready to evaluate performance of VBPR. The [BPR](https://arxiv.org/ftp/arxiv/papers/1205/1205.2618.pdf) model is also included as a baseline to examine the effectiveness of the text auxiliary data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vbpr = cornac.models.VBPR(k=10, k2=10, n_epochs=20, batch_size=20, learning_rate=0.001,\n",
```

Comment:
The hyperparameters (k, and n_epochs/max_iter) for VBPR and BPR are different. Can we choose these parameters in such away that VBPR collapses to BPR if remove the visual features? At least have the same `k`, it is fine to have different  `n_epochs/max_iter` since we are dealing with different objective functions.

Reply:
Or may be the objective here is to make `k_vbpr+k2_vbpr = k_bpr` ?

Reply:
Yes, it is. Anyway, I updated the tutorial.
Do you have any other comments?